airflow/dag_processing/processor.py

```diff
@@ -365,8 +365,10 @@ def __init__(self, dag_ids: list[str] | None, dag_directory: str, log: logging.L
         self._dag_directory = dag_directory
         self.dag_warnings: set[tuple[str, str]] = set()
 
+    @staticmethod
+    @internal_api_call
     @provide_session
-    def manage_slas(self, dag: DAG, session: Session = None) -> None:
+    def manage_slas(dag_folder, dag_id: str, log: logging.Logger, session: Session = NEW_SESSION) -> None:
```

Comment:
Passing in a logger object feels very wrong to me.

Reply:
I agree, it feels also very wrong to me but maybe with some context it will help understanding it :).

All methods decorated with `@internal_api_call` are meant to be called either directly (like today) or through a new component: internal API ([see PR here](https://github.com/apache/airflow/pull/27892)). This decision is based on the the flag `database_access_isolation`. What we want to achieve here is when the methods are called directly, the behavior should be the same as today, hence the logger should be the same. However, when these methods are called through the internal API, we want the logs to be separated from the current execution. Then, in that case we want a different logger. This new logger is created by the internal API itself, [see PR here](https://github.com/apache/airflow/pull/28476).

I hope it helps to understand and feel free to give me your thoughts given this context

Reply:
I understand the technical aspects, but it would simply indicate either the decorator should be improved to remove this restriction, or we need a new mechanism (perhaps built upon `@internal_api_call` that can be used on an instance method, not only static methods.

Reply:
@mhenc what do you think? Meanwhile I'll try to find a in-between solution

Reply:
I don't think we could make it work for non-static methods - passing the whole state of the object would be tricky (we want to avoid pickling) and would affect the performance even more.

If the only problem is passing the logger, then I can see few options:
1. Instead of passing the logger, just send the logger name and create the logger inside the method
2. Make logger static (as Vinc suggested in the first place) - we can wrap it into some method to keep `self.log` available for other methods.
3. I think internal_api_call should also work for class methods, so we could do something like this:
https://stackoverflow.com/a/8438818

cc: @potiuk

Reply:
Point 3 is probably best in the long run IMO.

Reply:
Do you mean like [this](https://github.com/apache/airflow/pull/28502/commits/14546b1c872104edb4fb7ab8e6bcedbbc2767fc1)? I could not find a way to remove the duplicate methods of `log` and `get_log` in `logging_mixin.py`. If you have a solution, I'd be very happy to use it :)

Reply:
See the method `get_log` I added [here](https://github.com/apache/airflow/pull/28502/files#diff-af25dd96233a97ace811c614fa7d5bd059cbbc1571e421f6675e16f6290814c5R71). It is literally a duplicate of the property log defined just below

Reply:
I see. challenge accepted

Reply:
PR here: https://github.com/aws-mwaa/upstream-to-airflow/pull/3/files

Reply:
Thanks! I'll update this PR accordingly

Reply:
Not nice bot.. Closed my PR :D

Reply:
Hahaha, yeah! It is a bossy and rude bot xD

Reply:
PR updated